switch/clock.ml

```diff
@@ -14,20 +14,14 @@
  * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  *)
 
-let use_mtime () = Mtime.Span.to_uint64_ns (Mtime_clock.elapsed ())
+include Mtime_clock
 
-let use_timeofday () = Int64.of_float (Unix.gettimeofday () *. Mtime.s_to_ns)
+type +'a io = 'a Lwt.t
 
-let ns =
-  try
-    Mtime_clock.now () |> ignore;
-    use_mtime
-  with Sys_error e -> begin
```

Comment:
From `mtime` documentation:

>  The functions Mtime_clock.elapsed, Mtime_clock.now, Mtime_clock.counter, Mtime_clock.elapsed_ns and Mtime_clock.now_ns raise Sys_error whenever they can't determine the current time or that it doesn't fit in Mtime's range. Ususally this exception should only be catched at the toplevel of your program to log it and abort the program. It indicates a serious error condition in the system.

> All the other functions, whose functionality is less essential, simply silently return None if they can't determine the information either because it is unavailable or because an error occured.

Reply:
I believe removing this check and let the program fail is the right thing to do here. Note that this did not create issues in any of the tests.

Reply:
Ping @robhoes @jonludlam